### PR TITLE
Fix #1400 : Adding prow style OWNERS files

### LINKED ISSUE #1400: Add prow style OWNERS to odo repository
[kind/Enhancement]

## Which functionality do you think we should update/improve?
Add appropriate OWNERS files to odo repository (ref : OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md).

Files can be declared at any level. Eg, we can have global owners file that will apply to entire repo, and child owners which will apply to particular directory

## Why is this needed? 
Prow has plugins which mostly used this to decide who needs to be pinged (blunderbuss plugin) and whose approvals count as approvals for automatic merge (tide) .

This will make it easier to integrate with OpenShift prow automated workflow.

Ref : https://github.com/redhat-developer/odo/issues/1167

### PR DESCRIPTION
**Pull request by mohammedzee1000**
## What is the purpose of this change? What does it change?

This PR adds OWNERS files to enable prow to ping appropriate people
and decide whose PRs warrant automatic merge

## Was the change discussed in an issue?
fixes #1400 

Signed-off-by: Mohammed Zeeshan Ahmed <mohammed.zee1000@gmail.com>

### CODE REVIEW

**Comment by kadel** [comment]
Does it make sense to have everyone in the top level OWNERS?
Does the "lower level" OWNERS has precedence over toplevel?
Does it make sense to have everyone as a reviewer and also as an approver? Isn't every approver automatically also reviewer? 

It would be good to remove github CODEOWNERS so it doesn't conflict

File: tests/OWNERS [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# See the OWNERS docs: https://git.k8s.io/community/contributors/guide/owners.md
+
+approvers:
+- kadel
```
Comment (kadel) [inline tests/OWNERS]: please remove me from here ;-)

Comment (mohammedzee1000) [reply tests/OWNERS]: Removed :)

**Comment by mohammedzee1000** [comment]
> Does it make sense to have everyone in the top level OWNERS?

The top level OWNERS is default ownership.

> Does the "lower level" OWNERS has precedence over toplevel?

Whenever a file is modified, it finds approvers from closest OWNERS at same level or above. So yes, it does.

> Does it make sense to have everyone as a reviewer and also as an approver? Isn't every approver automatically also reviewer?

It is what I have seen in most OWNERS files, I do need to check if its compulsory though
> 
> It would be good to remove github CODEOWNERS so it doesn't conflict

Yes, i guess it would. Do we want to do that now, in this PR or wait for openshift/release PR (not raised yet) to be merged?

**Comment by mohammedzee1000** [comment]
https://git.k8s.io/community/contributors/guide/owners.md
This should give overview of how it works in conjunction with tide

**Comment by kadel** [comment]
> > Does it make sense to have everyone in the top level OWNERS?
> 
> The top level OWNERS is default ownership.
> 
> > Does the "lower level" OWNERS has precedence over toplevel?
> 
> Whenever a file is modified, it finds approvers from closest OWNERS at same level or above. So yes, it does.

Sweet, that is what I hoped for.


> 
> > Does it make sense to have everyone as a reviewer and also as an approver? Isn't every approver automatically also reviewer?
> 
> It is what I have seen in most OWNERS files, I do need to check if its compulsory though

ok

> 
> > It would be good to remove github CODEOWNERS so it doesn't conflict
> 
> Yes, i guess it would. Do we want to do that now, in this PR or wait for openshift/release PR (not raised yet) to be merged?

Not sure that anyone is actually using codeowners right now, but you are right it will be cleaner to wait till we have this setup

**Comment by cdrage** [comment]
LGTM! :+1:

**Comment by metacosm** [comment]
Github codeowners are used to automatically assign people to review PRs but I'm not sure that people actually pay attention to this. :(

**Comment by girishramnani** [comment]
I always want to believe that things would improve :-)

**Comment by kadel** [comment]
> Github codeowners are used to automatically assign people to review PRs but I'm not sure that people actually pay attention to this. :(

Based on what I see they don't :-(

**Pull request merged**